src/pages/Scholarships.jsx

```diff
@@ -26,14 +29,41 @@ function ScholarshipsPage() {
   const [error, setError] = useState();
   const [sortField, setSortField] = useState('deadline');
   const [sortDir, setSortDir] = useState('asc');
+  const [amountFilterVals, setAmountFilterVals] = useState({ min: 0, max: 0 });
+  const location = useLocation();
+
+  useEffect(() => {
+    const params = queryString.parse(location.search);
+    const minParam = parseInt(params.min ?? 0, 10);
+    const maxParam = parseInt(params.max ?? 0, 10);
+    setAmountFilterVals({
+      min: minParam >= 0 ? minParam : 0,
+      max: maxParam >= 0 ? maxParam : 0,
```

Comment:
`minAmount` and `maxAmount` might in case there are other min/max filters in the future.